extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -309,9 +338,69 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
   }
 
   /** @override */
+  buildCallback() {
+    super.buildCallback();
+
+    if (this.isResponsive()) {
```

Comment:
I think calculations in the`onLayoutMeasure` and then request in `onLayoutCallback` was what we thought was the best indeed. That leaves when we actually do resize itself, and I believe the best place is in `onLayoutMeasure` as long as we avoid cycles. Keep in mind - by virtue of the browsing parse, a lot of times the size is not possible to tell in the `buildCallback`.